client/client.go

```diff
@@ -308,6 +314,15 @@ type (
 		//  - QueryFailError
 		QueryWorkflow(ctx context.Context, workflowID string, runID string, queryType string, args ...interface{}) (encoded.Value, error)
 
+		// QueryWorkflowV2 queries a given workflow execution and returns the query result synchronously.
+		// See QueryWorkflowV2Request and QueryWorkflowV2Response for more information.
+		// The errors it can return:
+		//  - BadRequestError
+		//  - InternalServiceError
+		//  - EntityNotExistError
+		//  - QueryFailError
+		QueryWorkflowV2(ctx context.Context, request QueryWorkflowV2Request) (QueryWorkflowV2Response, error)
```

Comment:
This is a really bad API name but I am not sure what to call it. Let's discuss offline.